Source/Core/Gen4/Generators/WildGenerator4.cpp

```diff
@@ -244,6 +244,160 @@ std::vector<WildState4> WildGenerator4::generateMethodJ(u32 seed, const Encounte
     return states;
 }
 
+template <bool bug>
+void createPokemon(PokeRNG &go, WildState4 &state, u8 buffer, u8 synchNature, u8 genderRatio, u16 tsv, const Lead &lead,
+                   const StateFilter &filter)
+{
+    u32 pid = 0;
+    if constexpr (bug)
+    {
+        bool cuteCharmSuccess = false;
```

Comment:
This variable shouldn't be necessary with the below change to the true section of cute charm